src/jimp2/wireworld/z8/wireworldlogic/Rules.java

```diff
@@ -5,42 +5,42 @@
 
     private int[] nearbyHeadsToBecomeHead;
 
+    public Rules ()
+    {
+        nearbyHeadsToBecomeHead= new int [2];
+        nearbyHeadsToBecomeHead[0]=1;
+        nearbyHeadsToBecomeHead[1]=2;
+    }
 
-    /**
-     * @param cell
-     */
     private void tailRule(Cell cell) {
-        // TODO implement here
+            cell.setState(State.CONDUCTOR);
     }
 
-    /**
-     * @param cell
-     */
     private void headRule(Cell cell) {
-        // TODO implement here
+            cell.setState(State.TAIL);
     }
 
-    /**
-     * @param cell 
-     * @param neighbouringHeads
-     */
     private void conductorRule(Cell cell, int neighbouringHeads) {
-        // TODO implement here
+        for(int i : nearbyHeadsToBecomeHead)
+        {
+            if(neighbouringHeads == i)
+            {
+                cell.setState(State.HEAD);
+            }
+        }
```

Comment:
1. Co w przypadku gdy warunek nie zostanie spełniony? Przydałby się względem tego jakiś komentarz.
2. Nazwanie zmiennej "i" w tym kontekście jest nieco mylące, bo nie jest ona tu iteratorem a kolejnymi możliwościami spełnienia danego warunku.

Reply:
Ad.1) Jeśli nie jest spełnione to nie zmienia się. Wydawało mi się to oczywiste